src/modules/upower.rs

```diff
@@ -185,7 +188,18 @@ impl Module<gtk::Button> for UpowerModule {
 
         let rx = context.subscribe();
         glib_recv!(rx, properties => {
-            let format = format.replace("{percentage}", &properties.percentage.to_string());
+            let state = properties.state;
+            let is_charging = state == BatteryState::Charging || state == BatteryState::PendingCharge;
+            let time_remaining;
+            if is_charging {
+                time_remaining = seconds_to_string(properties.time_to_full);
+            }
+            else {
+                time_remaining = seconds_to_string(properties.time_to_empty);
+            }
```

Comment:
In Rust `if else` is actually an expression, so you can use it directly for assignment:

```suggestion
            let time_remaining = if is_charging {
                seconds_to_string(properties.time_to_full);
            }
            else {
                seconds_to_string(properties.time_to_empty);
            }
```

(I don't think that suggestion is `rustfmt` compliant so you'll probs need to run after applying)

Reply:
Sadly this doesn't work, as then the `time_remaining` variable will be of type `&()`

Reply:
My bad - the semicolons make the difference there due to Rust's implicit returns. Updated.

Reply:
Implemented the change.